node/src/main/scala/com/wavesplatform/settings/package.scala

```diff
@@ -44,35 +44,53 @@ package object settings {
   }
 
   def loadConfig(maybeUserConfig: Option[Config]): Config = {
-    val directoryDefaults = ConfigFactory
-      .parseString(s"waves.directory = $defaultDirectory")
-
     val defaults = ConfigFactory.defaultOverrides()
-
-    maybeUserConfig
+    val external = maybeUserConfig
       .fold(defaults)(defaults.withFallback)
+
+    val withApp = external
+      .withFallback(ConfigFactory.defaultApplication())
+
+    val directoryDefaults = ConfigFactory
+      .parseString(s"waves.directory = ${defaultDirectory(withApp)}")
+
+    external
       .withFallback(directoryDefaults)
       .withFallback(ConfigFactory.defaultApplication())
       .withFallback(ConfigFactory.defaultReference())
       .resolve()
   }
 
-  def defaultDirectory: String =
-    if (SystemUtils.IS_OS_WINDOWS) winDefaultDirectory
-    else if (SystemUtils.IS_OS_MAC) osxDefaultDirectory
-    else nixDefaultDirectory
+  def defaultDirectory(config: Config): String = {
+    // No actual interpolation here, `s` to suppress warnings
+    def osxDefaultDirectory: String =
+      s"$${user.home}/Library/Application Support"
+
+    //noinspection SpellCheckingInspection
+    def winDefaultDirectory: String =
+      s"$${LOCALAPPDATA}"
+
+    def nixDefaultDirectory: String = {
+      val maybeXdgDir = sys.env.get("XDG_DATA_HOME")
+      val defaultDir = s"$${user.home}/.local/share"
+
+      maybeXdgDir getOrElse defaultDir
+    }
 
-  // No actual interpolation here, `s` to suppress warnings
-  def osxDefaultDirectory: String =
-    s"$${user.home}/Library/Application Support/waves"
+    def withNetwork(config: Config): String = {
+      val bc = config.getString("waves.blockchain.type")
+      val suffix =
+        if (bc == "CUSTOM") s"custom${config.getString("waves.blockchain.custom.address-scheme-character")}"
```

Comment:
This approach has several flaws: some filesystems are case-insensitive, some forbid special charaters in directory names. A safer way to support arbitrary characters would be to append decimal or hexadecimal value as a suffix (e.g. `waves-custom-0xff`)